shell/platform/tizen/channels/platform_view_channel.cc

```diff
@@ -107,130 +92,193 @@ void PlatformViewChannel::HandleMethodCall(
-        !GetValueFromEncodableMap(arguments, "id", &view_id) ||
-        !GetValueFromEncodableMap(arguments, "width", &width) ||
-        !GetValueFromEncodableMap(arguments, "height", &height)) {
-      result->Error("Invalid arguments");
-      return;
-    }
+    OnCreate(arguments, std::move(result));
+  } else if (method == "clearFocus") {
+    OnClearFocus(arguments, std::move(result));
+  } else if (method == "dispose") {
+    OnDispose(arguments, std::move(result));
+  } else if (method == "resize") {
+    OnResize(arguments, std::move(result));
+  } else if (method == "touch") {
+    OnTouch(arguments, std::move(result));
+  } else {
+    FT_LOG(Warn) << "Unimplemented method: " << method;
+    result->NotImplemented();
+  }
+}
 
-    FT_LOG(Info) << "Creating a platform view: " << view_type;
-    RemoveViewInstanceIfNeeded(view_id);
+void PlatformViewChannel::OnCreate(
+    const EncodableValue* arguments,
+    std::unique_ptr<MethodResult<EncodableValue>>&& result) {
+  auto map_ptr = std::get_if<EncodableMap>(arguments);
+  if (!map_ptr) {
+    result->Error("Invalid arguments");
+    return;
+  }
 
-    EncodableMap values = std::get<EncodableMap>(arguments);
-    EncodableValue value = values[EncodableValue("params")];
-    ByteMessage byte_message;
-    if (std::holds_alternative<ByteMessage>(value)) {
-      byte_message = std::get<ByteMessage>(value);
-    }
-    auto it = view_factories_.find(view_type);
-    if (it != view_factories_.end()) {
-      auto focused_view = view_instances_.find(CurrentFocusedViewId());
-      if (focused_view != view_instances_.end()) {
-        focused_view->second->SetFocus(false);
-      }
-
-      auto view_instance =
-          it->second->Create(view_id, width, height, byte_message);
-      if (view_instance) {
-        view_instances_.insert(
-            std::pair<int, PlatformView*>(view_id, view_instance));
-        result->Success(EncodableValue(view_instance->GetTextureId()));
-      } else {
-        result->Error("Can't create a webview instance!!");
-      }
-    } else {
-      FT_LOG(Error) << "Can't find view type: " << view_type;
-      result->Error("Can't find view type");
-    }
-  } else if (method == "clearFocus") {
-    int view_id = -1;
-    if (std::holds_alternative<int>(arguments)) {
-      view_id = std::get<int>(arguments);
-    };
-    auto it = view_instances_.find(view_id);
-    if (view_id >= 0 && it != view_instances_.end()) {
-      it->second->SetFocus(false);
-      it->second->ClearFocus();
-      result->Success();
-    } else {
-      result->Error("Can't find view id");
+  EncodableMapValueGetter<std::string> view_type(map_ptr, "viewType");
+  EncodableMapValueGetter<int> view_id(map_ptr, "id");
+  EncodableMapValueGetter<double> width(map_ptr, "width");
+  EncodableMapValueGetter<double> height(map_ptr, "height");
+
+  if (!view_type.value || !view_id.value || !width.value || !height.value) {
```

Comment:
I'm not familiar with operator overloads, but do we need `.value`s here?

Reply:
You're right, my mistake.